pkg/util/taints/taints.go

```diff
@@ -288,15 +288,13 @@ func TaintKeyExists(taints []v1.Taint, taintKeyToMatch string) bool {
 func TaintSetDiff(t1, t2 []v1.Taint) (taintsToAdd []*v1.Taint, taintsToRemove []*v1.Taint) {
 	for _, taint := range t1 {
 		if !TaintExists(t2, &taint) {
-			t := taint
-			taintsToAdd = append(taintsToAdd, &t)
+			taintsToAdd = append(taintsToAdd, &taint)
```

Comment:
This was intentional.

Here for reference:
https://medium.com/swlh/use-pointer-of-for-range-loop-variable-in-go-3d3481f7ffc9